src/System.Console/src/System/ConsolePal.Unix.cs

```diff
@@ -63,7 +63,8 @@ public static ConsoleColor BackgroundColor
 
         public static void ResetColor()
         {
-            if (!ConsoleOutIsTerminal)
+            // We only want to reset colors if we're targeting a TTY device
+            if (Console.IsOutputRedirected)
```

Comment:
>We actually want the IsXxRedirected behavior to be the same across the platforms, right? In which case SetOut should not affect it on Unix either?

I thought we intentionally do this on Unix? Why do we make the Color behave this way? On windows even Color is based on standard input, output and error.